sdk/core/src/main/java/com/klaviyo/core/config/KlaviyoConfig.kt

```diff
@@ -108,7 +108,14 @@ object KlaviyoConfig : Config {
         private set
     override lateinit var sdkName: String private set
     override lateinit var sdkVersion: String private set
-    override lateinit var apiKey: String private set
+    private var _apiKey: String = ""
+    override var apiKey: String
+        get() = _apiKey.ifEmpty {
+            throw MissingAPIKey()
+        }
+        set(value) {
+            _apiKey = value
+        }
```

Comment:
This is kinda the key change for the whole PR: instead of throwing `MissingAPIKey` on `build()`, I'm now allowing a Config to be built without API key (so that Context can be captured). To maintain that API key validation for callers, I'm still throwing `MissingAPIKey` if you try to _access_ or _set_ an empty API key.